scripts/api.js

```diff
@@ -0,0 +1,116 @@
+import PocketChange from './pocket-change.js';
+import MacroSupport from './macro-support.js';
+import Settings from './settings.js';
+
+const API = {
+	PocketChange: undefined,
+    MacroSupport: undefined,
+
+    /**
+    * For all selected tokens, generate currency for them
+	*
+	* @param {boolean} - (optional) if true it will generate a random currency without the rating check
+    */
+	generateCurrencyForSelectedTokens(ignoreRating = false) {
+		if(!this.MacroSupport){
+			this.MacroSupport = new MacroSupport();
+		}
+		if(!this.PocketChange){
+			this.PocketChange = new PocketChange();
+		}
+		this.MacroSupport.prototype.generateCurrencyForSelectedTokens(ignoreRating);
+	},
+
+	/**
+	* For all selected tokens, convert them to lootable sheets.
+	*
+	* Adapted from the convert-to-lootable.js by @unsoluble, @Akaito, @honeybadger, @kekilla, and @cole.
+	*
+	* @param {number} chanceOfDamagedItems (optional) A number between 0 and 1 that corresponds to the percent chance an item will be damaged
+	* @param {number} damagedItemsMultiplier (optional) A number between 0 and 1 that will lower a damaged items value
+	* @param {boolean} removeDamagedItems (optional) If true, damaged items will be removed from the token rather than marked as damaged
+	*/
+	convertSelectedTokensToLootSheet(
+		chanceOfDamagedItems,
+		damagedItemsMultiplier,
+		removeDamagedItems
+	) {
+    if(!game.modules.get('lootsheet-simple')?.active) {
+      let word = 'install and activate';
+      if (game.modules.get('lootsheet-simple')) word = 'activate';
+      const errorText = `${Settings.PACKAGE_NAME} | Requires the 'lootsheet-simple' module. Please ${word} it.`.replace("<br>", "\n");
+      ui.notifications?.error(errorText);
+      throw new Error(errorText);
+    }
+		if(!this.MacroSupport){
+			this.MacroSupport = new MacroSupport();
+		}
+		if(!this.PocketChange){
+			this.PocketChange = new PocketChange();
+		}
+		this.MacroSupport.prototype.convertSelectedTokensToLoot({
+			chanceOfDamagedItems,
+			damagedItemsMultiplier,
+			removeDamagedItems,
+			mode: "lootsheet"});
+	},
+
+	/**
+	* For all selected tokens, convert them back from lootable sheets.
+	*/
+	revertSelectedTokensFromLootSheet() {
+    if(!game.modules.get('lootsheet-simple')?.active) {
+      let word = 'install and activate';
+      if (game.modules.get('lootsheet-simple')) word = 'activate';
+      const errorText = `${Settings.PACKAGE_NAME} | Requires the 'lootsheet-simple' module. Please ${word} it.`.replace("<br>", "\n");
+      ui.notifications?.error(errorText);
+      throw new Error(errorText);
+    }
+		if(!this.MacroSupport){
+			this.MacroSupport = new MacroSupport();
+		}
+		if(!this.PocketChange){
+			this.PocketChange = new PocketChange();
+		}
+		this.MacroSupport.prototype.revertSelectedTokensFromLootSheet();
```

Comment:
Since all of these methods are just calling `MacroSupport` methods... `MacroSupport` could probably just be coverted to the `API`. That's kind of what it was for initially anyway. It was the API when using macros

Reply:
I think the change should be to just move the code from `MacroSupport` into `API`, and remove most of this extra stuff in `API` that you have.

Reply:
done